assets/js/base/components/dropdown-selector/index.js

```diff
@@ -182,11 +184,20 @@ const DropdownSelector = ( {
 							getItemProps={ getItemProps }
 							getMenuProps={ getMenuProps }
 							highlightedIndex={ highlightedIndex }
-							options={ options.filter(
-								( option ) =>
+							options={ options.filter( ( option ) => {
+								if ( isCaseSensitive ) {
+									return (
+										! inputValue ||
+										option.name.includes( inputValue )
+									);
+								}
+								return (
 									! inputValue ||
-									option.value.startsWith( inputValue )
-							) }
+									option.name
+										.toLowerCase()
+										.includes( inputValue.toLowerCase() )
+								);
+							} ) }
```

Comment:
Could we dry this bit a bit? we could simplify it to normalize the inputValue and optionName only when needed ie.

we could also add trim to the inputValue

ie.

```
const optionName = isCaseSensitive ? option.name : option.name.toLowerCase();
const nameQuery = isCaseSensitive ? inputValue...

return ! inputValue || optionName.includes( nameQuery.trim() )
```

This way we'll have input/data normalisation separated and all business logic in one place